### PR TITLE
Fix some socket related bugs and add IPv6 support

### PR DESCRIPTION
**Pull request by jessehui**
Requires:

- [x] #171

### CODE REVIEW

File: demos/redis/run_occlum_redis.sh [hunk 1]
```diff
@@ -18,4 +18,7 @@ copy_bom -f $bomfile --root image --include-dir /opt/occlum/etc/template
 
 occlum build
 # 3. Run redis server
-occlum run /bin/redis-server /etc/redis.conf &
+# Default IPv6:
+occlum run /bin/redis-server --save "" --appendonly no &
```
Comment (guzongmin) [inline demos/redis/run_occlum_redis.sh:22]: Maybe comment the ipv4 in redis.conf is better than change the execution command.

File: src/libos/crates/host-socket/src/ioctl/get_ifreq.rs [hunk 2]
```diff
@@ -40,24 +41,48 @@ impl GetIfReqWithRawCmd {
...
-fn get_ifreq_by_host(fd: HostFd, cmd: u32) -> Result<IfReq> {
-    let mut if_req: IfReq = Default::default();
-    try_libc!(do_ioctl(
-        fd as _,
-        cmd as _,
-        &mut if_req as *mut IfReq as *mut i32
-    ));
+fn get_ifreq_by_host(fd: HostFd, cmd: u32, req: &IfReq) -> Result<IfReq> {
+    let mut if_req: IfReq = req.clone();
+    // info!("if_req = {:?}, fd = {:?}", if_req, fd);
```
Comment (guzongmin) [inline src/libos/crates/host-socket/src/ioctl/get_ifreq.rs:53]: Maybe we could remove these comments

File: demos/redis/run_occlum_redis.sh (hunk 1 above)
Comment (jessehui) [reply demos/redis/run_occlum_redis.sh]: Done.

File: src/libos/crates/host-socket/src/ioctl/get_ifreq.rs (hunk 2 above)
Comment (jessehui) [reply src/libos/crates/host-socket/src/ioctl/get_ifreq.rs]: Done. Thanks for your review.

**Review by guzongmin — APPROVED** [review]
LGTM

**Pull request merged**